roles/neutron-data-network/tasks/main.yml

```diff
@@ -110,6 +117,7 @@
   when: neutron.lbaas.enable
 
 - include: ipchanged.yml
+  when: not neutron.l3ha.enabled|bool
```

Comment:
I would leave ipchanged in place, but modify `migrate_neutron_services` to skip migrating HA neutron routers.